File: src/main/java/com/balatro/Cache.java
package com.balatro;

import org.jetbrains.annotations.NotNull;

public interface Cache {

    Double get(String id);

    void setGeneratedFirstPack(boolean generatedFirstPack);

    boolean isGeneratedFirstPack();

    void put(String id, double value);

    double get(@NotNull Coordinate c);

    void put(@NotNull Coordinate c, double value);
}

File: docs/js/immolate2/src/utils/Cache.js
export class Cache {
    constructor() {
        this.nodes = new Map();
        this.generatedFirstPack = false;
    }

    isGeneratedFirstPack() {
        return this.generatedFirstPack;
    }

    setGeneratedFirstPack(generatedFirstPack) {
        this.generatedFirstPack = generatedFirstPack;
    }

    getNode(key) {
        return this.nodes.get(key);
    }

    setNode(key, value) {
        this.nodes.set(key, value);
    }
}